File: test_cases/python/claude/07.py
def run_tests(solution_module):
    f = solution_module.isMatch
    # multiple ? chars
    assert f('abc', '???') is True
    # star at the end consumes remainder
    assert f('abcdef', 'abc*') is True
    # consecutive stars behave like one
    assert f('abc', '***') is True
    # empty string, empty pattern
    assert f('', '') is True
    # star cannot match if pattern requires more
    assert f('ab', 'a*c') is False

File: test_cases/c/claude/07.c
#include <stdio.h>
#include <stdbool.h>
#include <assert.h>

bool isMatch(char* s, char* p);

int main(void) {
    assert(isMatch("aa", "a") == false);
    assert(isMatch("aa", "*") == true);
    assert(isMatch("cb", "?a") == false);
    assert(isMatch("", "") == true);
    assert(isMatch("adceb", "*a*b") == true);
    return 0;
}